api/services/search.service.js

```diff
@@ -0,0 +1,16 @@
+import { InvalidData } from "../errors/InvalidData.js";
+import * as searchRepository from "../repositories/search.repository.js";
+export const getProductsAndStores = async (req, res, next) => {
+    var { query } = req.body;
+    if(query.length < 3){
+        return next(InvalidData.invalidQuery());
+    }
+    query = query.toString();
+    const searchedData = await searchRepository.getProductsAndStores(query);
+    res.status(200).json(searchedData);
```

Comment:
what does the return type look like? do we need to map all these to some new object structure?

Reply:
It returns { products:{}, stores: {} }

Reply:
JSON object?